rubrik_cdm/cluster.py

```diff
@@ -606,3 +608,56 @@ def read_only_authorization(self, username, timeout=15):
 
         self.log("read_only_authorization: Granting read-only privilages to user '{}'.".format(username))
         return self.post("internal", "/authorization/role/read_only_admin", config, timeout)
+
+    def cluster_node_id(self,timeout=15):
+        """ Returns a list of node ids from all the nodes in the cluster.
+
+                Arguments:
+                    None
+
+                Keyword Arguments:
+                    None
+
+                Returns:
+                    dict -- The full API response from `POST /internal/node`.
+                """
+        self.log("cluster_node_id - Getting all the node ids from the from the cluster")
+        node_id = self.get('internal', '/node',timeout)
+        return node_id
+
+    def cluster_support_tunnel(self,enabled, timeout=15):
+        """ Function that can check the status, enable or disable the support tunnel.
+
+                        Arguments:
+                            enabled - with state of (True or False)
+
+                        Keyword Arguments:
+
+                            enabled with state of True checks the status first then enables the support tunnel if needed
+                            enabled with state of False will disable the current tunnel
+
+                        Returns:
+                            dict -- The full API response from `POST /internal/node/me/support_tunnel`.
+                        """
+        if enabled not in [True, False]:
+            raise InvalidParameterException("The action parameter must be status, enable, or disable")
+
+        if enabled is True:
+            self.log("cluster_support_tunnel - Check Status of Cluster Support Tunnel and Enable it if needed")
+            check_tunnel = self.get('internal', '/node/me/support_tunnel', timeout)
+            if check_tunnel['isTunnelEnabled'] is False:
+                config = {}
+                config['isTunnelEnabled'] = True
+                config['inactivityTimeoutInSeconds'] = 0
+                tunnel = self.patch('internal', '/node/me/support_tunnel', config,timeout)
+                return tunnel
+
+            else:
+                return("Tunnel is already enabled")
+
+        elif enabled is False:
+            self.log("cluster_support_tunnel - Disable the Support Tunnel")
+            config = {}
+            config['isTunnelEnabled'] = False
+            tunnel = self.patch('internal', '/node/me/support_tunnel', config,timeout)
```

Comment:
Need a log message saying that we are enabling/disabling the tunnel before sending the API call